test/integration/targets/lookup_first_found/tasks/main.yml

```diff
@@ -71,3 +71,33 @@
   assert:
     that:
       - "this_not_set is not defined"
+
+# https://github.com/ansible/ansible/issues/70772
+- name: test first_found with files with empty variables works (dict)
+  set_fact:
+    file_found_dict_var: "{{ item }}"
+  with_first_found:
+    - files:
+        - "{{ this_var_is_empty }}"
+        - "{{ role_path + '/files/foo1' }}"
+  vars:
+    this_var_is_empty: {}
```

Comment:
Does this replicate the behavior the user reported?  I would either not define `this_var_is_empty` at all, or do `this_var_is_empty.foo` without `foo` defined.

Reply:
even with this is triggering the bug from the test I did (that why I commited first the test and then the fix) but I can do this method too.